src/equations/linearized_euler_1d.jl

```diff
@@ -0,0 +1,143 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    LinearizedEulerEquations1D(v_mean_global, c_mean_global, rho_mean_global)
+
+Linearized euler equations in one space dimension. The equations are given by
+```math
+\partial_t
+\begin{pmatrix}
+    \rho' \\ v_1' \\ p'
+\end{pmatrix}
++
+\partial_x
+\begin{pmatrix}
+    \bar{\rho} v_1' + \bar{v_1} \rho ' \\ \bar{v_1} v_1' + \frac{p'}{\bar{\rho}} \\ \bar{v_1} p' + c^2 \bar{\rho} v_1'
+\end{pmatrix}
+=
+\begin{pmatrix}
+    0 \\ 0 \\ 0
+\end{pmatrix}
+```
+The bar ``\bar{(\cdot)}`` indicates uniform mean flow variables and c is the speed of sound.
+The unknowns are the acoustic velocity ``v_1'``, the pressure ``p'`` and the density ``\rho'``.
+"""
+struct LinearizedEulerEquations1D{RealT <: Real} <:
+       AbstractLinearizedEulerEquations{1, 3}
+    v_mean_global::RealT
+    c_mean_global::RealT
+    rho_mean_global::RealT
+end
+
+function LinearizedEulerEquations1D(v_mean_global::Real,
+                                    c_mean_global::Real, rho_mean_global::Real)
+    if rho_mean_global < 0
+        throw(ArgumentError("rho_mean_global must be non-negative"))
+    elseif c_mean_global < 0
+        throw(ArgumentError("c_mean_global must be non-negative"))
+    end
+
+    return LinearizedEulerEquations1D(v_mean_global, c_mean_global,
+                                      rho_mean_global)
+end
+
+# Constructor with keywords (note the leading ';')
```

Comment:
Do we note this elsewhere? seems like overkill, although if this equation system is intended for teaching and/or tutorial purposes I am not against it.

Reply:
I am not sure, I added this comment mostly for myself because I kept scrolling back and forth for 5 minutes to understand why the 2D linearized Euler equations have two seemingly identical constructors.